### PR TITLE
Move from ObjectBox to Isar

### PR DESCRIPTION
**Pull request by vishnukvmd**

### CODE REVIEW

File: lib/db/embeddings_db.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  Future<List<Embedding>> getUnsyncedEmbeddings() async {
+    return await _isar.embeddings.filter().updationTimeEqualTo(null).findAll();
+  }
+
+  Future<void> deleteAllForModel(Model model) async {
+    await _isar.writeTxn(() async {
+      final embeddings =
+          await _isar.embeddings.filter().modelEqualTo(model).findAll();
+      await _isar.embeddings.deleteAll(embeddings.map((e) => e.id).toList());
```
Comment (0nullpointer) [inline lib/db/embeddings_db.dart:55]: Do you need an embedding deleted event similar to `EmbeddingUpdatedEvent`?

File: lib/db/embeddings_db.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  static final EmbeddingsDB instance = EmbeddingsDB._privateConstructor();
+
+  Future<void> init() async {
+    final dir = await getApplicationDocumentsDirectory();
+    _isar = await Isar.open(
+      [EmbeddingSchema],
+      directory: dir.path,
+    );
+    // TODO: _clearDeprecatedStore(dir);
```
Comment (0nullpointer) [inline lib/db/embeddings_db.dart:22]: Remove?

**Review by 0nullpointer — APPROVED** [review]

File: lib/db/embeddings_db.dart (hunk 1 above)
Comment (vishnukvmd) [reply lib/db/embeddings_db.dart:55]: Yep.

File: lib/db/embeddings_db.dart (hunk 2 above)
Comment (vishnukvmd) [reply lib/db/embeddings_db.dart]: Was hedging, uncommented, will drop the ObjectBox DB.

**Review by 0nullpointer — APPROVED** [review]

**Comment by licaon-kter** [comment]
So users get migrated from objectbox to isar or how?

Note that F-Droid needs to build isar libs and will reject prebuild .so's :wink:

**Comment by vishnukvmd** [comment]
@licaon-kter we're dropping the data stored in ObjectBox, pulling it from the server, and pushing to Isar.

Will figure out a way to build Isar from source, thanks for the heads up!

**Pull request merged**